test/built-ins/Temporal/TimeZone/prototype/id/custom-timezone.js

```diff
@@ -0,0 +1,28 @@
+// Copyright (C) 2020 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-get-temporal.timezone.prototype.id
+description: Getter calls toString() and returns its value
+includes: [compareArray.js, temporalHelpers.js]
+features: [Temporal]
+---*/
+
+const actual = [];
+const expected = [
+  "get [Symbol.toPrimitive]",
+  "get toString",
+  "call timeZone.toString",
+];
+
+const timeZone = new Temporal.TimeZone("UTC");
+TemporalHelpers.observeProperty(actual, timeZone, Symbol.toPrimitive, undefined);
+TemporalHelpers.observeProperty(actual, timeZone, "toString", function () {
+  actual.push("call timeZone.toString");
+  return "time zone";
+});
+
+const descriptor = Object.getOwnPropertyDescriptor(Temporal.TimeZone.prototype, "id");
+const result = descriptor.get.call(timeZone);
```

Comment:
Fair enough. I originally kept it this way because otherwise it would push another property lookup onto the `actual` array, but that's no longer the case now that we cannot use a Proxy.